### PR TITLE
Fixup of general tls record parsing functions for dtls 1.3

### PR DESCRIPTION
**Pull request by fwh-dc**

### CODE REVIEW

**Comment by fwh-dc** [comment]
Ci failures look irrelevant

File: ssl/record/methods/tls_common.c [hunk 1]
```diff
@@ -1700,7 +1700,7 @@ int tls_post_encryption_processing_default(OSSL_RECORD_LAYER *rl,
         rl->msg_callback(1, thiswr->rec_version, SSL3_RT_HEADER, recordstart,
                          headerlen, rl->cbarg);
 
-        if (rl->version == TLS1_3_VERSION && rl->enc_ctx != NULL) {
+        if ((rl->version == TLS1_3_VERSION || rl->version == DTLS1_3_VERSION) && rl->enc_ctx != NULL) {
```
Comment (t8m) [inline ssl/record/methods/tls_common.c:1703]: Please split the line

**Review by t8m — CHANGES_REQUESTED** [review]

**Review by t8m — APPROVED** [review]

**Review by mattcaswell — APPROVED** [review]

**Review by t8m — APPROVED** [review]
@mattcaswell please reconfirm

**Review by mattcaswell — APPROVED** [review]

**Comment by t8m** [comment]
Merged to the feature branch. Thank you for your contribution.

**Pull request closed without merging**